core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -412,7 +418,7 @@ public void onLeadershipChange(Set<Partition> partitionsBecomeLeader,
                                    Map<String, Uuid> topicIds) {
         LOGGER.debug("Received leadership changes for leaders: {} and followers: {}", partitionsBecomeLeader, partitionsBecomeFollower);
 
-        if (this.rlmConfig.isRemoteStorageSystemEnabled() && !isRemoteLogManagerConfigured()) {
+        if (config.remoteLogManagerConfig().isRemoteStorageSystemEnabled() && !isRemoteLogManagerConfigured()) {
```

Comment:
this is unrelated to this PR, but `config.remoteLogManagerConfig` always return the same config even though the `config` is updated dynamically. That seems to be error-prone, since it means the "updated" config can return "remoteLogManagerConfig" with "previous" configs.

Reply:
yes, this is the main issue. We access the remote configurations using `KafkaConfig.remoteLogManagerConfig.xyz()` but it reflects to the config in the static file not the dynamically updated one.

Reply:
As we have great `RemoteLogManagerConfig` for remote storage, could we avoid moving configs out of `RemoteLogManagerConfig`? Especially, we are trying to reduce the size of `KafkaConfigs`.

If this PR aims to fix it for zk mode, maybe we can make sure the `remoteLogManagerConfig` returned by `config`  always has the latest configs. For example:

1.  `config.remoteLogManagerConfig` always create new `remoteLogManagerConfig` based on `currentConfig`
2. `KafkaConfig#updateCurrentConfig` should update inner `_remoteLogManagerConfig` also

@kamalcph WDYT?

Reply:
Can we take this refactoring as part of KAFKA-16976 ticket?

Reply:
@chia7712 Good point. A similar issue is raised in the [comment](https://github.com/apache/kafka/pull/16353#pullrequestreview-2121953295). We can fix the bug with the current approach in 3.8 and finish cleaning up the raised issues in a followup as part of KAFKA-16976

Reply:
@kamalcph @satishd sounds good to me :)